packages/vite/src/node/packages.ts

```diff
@@ -81,10 +82,22 @@ export function resolvePackageData(
 export function loadPackageData(
   pkgPath: string,
   preserveSymlinks?: boolean,
-  packageCache?: PackageCache
+  packageCache?: PackageCache,
+  cjsInclude?: (string | RegExp)[]
 ): PackageData {
   if (!preserveSymlinks) {
+    const originalPkgPath = pkgPath
     pkgPath = fs.realpathSync.native(pkgPath)
+
+    // In case a linked package is a local clone of a CommonJS dependency,
+    // we need to ensure @rollup/plugin-commonjs analyzes the package even
+    // after it's been resolved to its actual file location.
+    if (cjsInclude && pkgPath !== originalPkgPath) {
+      const filter = createFilter(cjsInclude, undefined, { resolve: false })
+      if (!filter(pkgPath) && filter(originalPkgPath)) {
+        cjsInclude.push(path.dirname(pkgPath) + '/**')
```

Comment:
Possible downside here is that `node_modules` might be inadvertently matched? It's not easy to determine the source folder in a package, so might have to live with it.